ios/brave-ios/Sources/Brave/Frontend/Shields/ShieldsPanelView.swift

```diff
@@ -0,0 +1,413 @@
+import Strings
+import SwiftUI
+
+struct ShieldsPanelView: View {
+  enum Action {
+    enum NavigationTarget {
+      case shareStats
+      case reportBrokenSite
+      case globalShields
+    }
+    case navigate(NavigationTarget, dismiss: Bool)
+    case changedShieldSettings
+    case shredSiteData
+  }
+
+  private let url: URL
+  private let displayHost: String
+  @State private var favicon = Image(uiImage: Favicon.defaultImage)
+  @AppStorage("advancedShieldsExpanded") private var advancedShieldsExpanded = false
+  @ObservedObject private var viewModel: ShieldsSettingsViewModel
+  private var actionCallback: (Action) -> Void
+
+  @MainActor init(url: URL, tab: Tab, domain: Domain, callback: @escaping (Action) -> Void) {
+    self.url = url
+    self.viewModel = ShieldsSettingsViewModel(tab: tab, domain: domain)
+    self.actionCallback = callback
+    self.displayHost = URLFormatter.formatURLOrigin(
+      forDisplayOmitSchemePathAndTrivialSubdomains: url.absoluteString
+    )
+  }
+
+  private var shieldsEnabledAccessibiltyLabel: String {
+    viewModel.shieldsEnabled
+      ? Strings.Shields.statusValueUp : Strings.Shields.statusValueDown
+  }
+
+  var body: some View {
+    ScrollView {
+      VStack(spacing: 16) {
+        headerView
+
+        if viewModel.shieldsEnabled {
+          shieldsReportView
+          Text(Strings.Shields.siteBroken)
+            .font(.caption)
+            .foregroundStyle(Color(.secondaryBraveLabel))
+            .multilineTextAlignment(.leading)
+            .padding(.horizontal)
+          DisclosureGroup(isExpanded: $advancedShieldsExpanded) {
+            advancedShieldsSection
+          } label: {
+            Text(Strings.Shields.advancedControls)
+              .foregroundStyle(Color(.braveLabel))
+              .frame(maxWidth: .infinity, alignment: .leading)
+          }.disclosureGroupStyle(ShieldsPanelDisclosureStyle())
+        } else {
+          shieldsOffFooterView
+        }
+      }
+      .padding(.top)
+    }
+    .background(Color(.braveBackground))
+    .frame(idealWidth: 360, alignment: .center)
+    .task {
+      if let faviconImage = FaviconFetcher.getIconFromCache(for: url)?.image {
+        self.favicon = Image(uiImage: faviconImage)
+      }
+      await loadFavicon(for: url, isPrivateBrowsing: viewModel.isPrivateBrowsing)
+    }
+    .toolbar(.hidden)
+  }
+
+  @ViewBuilder @MainActor private var headerView: some View {
+    VStack(alignment: .center, spacing: 8) {
+      HStack(alignment: .center, spacing: 8) {
+        favicon
+          .resizable()
+          .aspectRatio(contentMode: .fit)
+          .cornerRadius(4)
+          .frame(width: 24, height: 24, alignment: .center)
+        Text(displayHost)
+          .font(.title2)
+          .foregroundStyle(Color(.bravePrimary))
+      }
+      .frame(minWidth: .zero, alignment: .center)
+
+      ShieldsSwitchView(isEnabled: $viewModel.shieldsEnabled)
+        .frame(
+          width: ShieldsSwitch.size.width,
```

Comment:
Why do we hard-code the size of the view here?

Reply:
just because its already hard-coded in that existing class so I wanted it to match. I don't remember if there were any specific reasons other than that but I can see what happens when I remove this